### PR TITLE
fix: `calculateTxChargeableBytes` when transaction has no witnesses

### PR DESCRIPTION
**Pull request by nedsalk**
The chargeable bytes of a transaction are all transaction bytes except the witnesses part of the transaction.
`calculateTxChargeableBytes` was calculating properly when witnesses were present, but not when there were none: it returned `0` instead of `transactionBytes.length`. `transactionBytes.length` is the correct value because, without witnesses, _all_ the bytes of a transaction are chargeable.

### CODE REVIEW

**Comment by nedsalk** [comment]
> @camsjams
>  Changes make sense, do you think we need more tests around this?

Not really, I think these are enough as far as the functionality goes. 
On a related note, there is also #1331 which will refactor out `Witness.offset` and these tests will be affected again.

File: packages/providers/src/utils/calculate-fee.test.ts [hunk 1]
```diff
@@ -120,33 +120,34 @@ describe(__filename, () => {
...
-          dataLength: 64,
-          data: '0x3d8943b87436d54d7f1282ed9b01d38633f8003a37676276bbaf0560390d21f826b5f6e9ea85b1bca435af2bec4982436c20c1294876025f78aaa0143867eb7f',
-          offset,
-        },
-      ];
 
       const chargeableBytesFee = calculateTxChargeableBytes({
         transactionBytes,
-        transactionWitnesses,
+        transactionWitnesses: [],
```
[the remark this answers is not in the document]
Comment (nedsalk) [reply packages/providers/src/utils/calculate-fee.test.ts:147]: It happens :man_shrugging:

File: packages/providers/src/utils/calculate-fee.ts [hunk 2]
```diff
@@ -31,9 +31,9 @@ export interface CalculateTxChargeableBytesParams {
 
 /** @hidden */
 export const calculateTxChargeableBytes = (params: CalculateTxChargeableBytesParams): BN => {
-  const { transactionWitnesses } = params;
+  const { transactionWitnesses, transactionBytes } = params;
 
-  const txChargeableBytes = bn(transactionWitnesses?.[0]?.offset || 0);
+  const txChargeableBytes = bn(transactionWitnesses?.[0]?.offset || transactionBytes.length);
```
Comment (Torres-ssf) [inline packages/providers/src/utils/calculate-fee.ts:36]: Good catch.

**Review by Torres-ssf — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

File: packages/providers/src/utils/calculate-fee.ts (hunk 2 above)
Comment (nedsalk) [reply packages/providers/src/utils/calculate-fee.ts:36]: Thanks!

**Pull request merged**